### ISSUE TITLE
no-suite-dupes should have nesting options

### ISSUE THREAD

**Issue by reintroducing** [tlvince/eslint-plugin-jasmine#6]
Thanks for this plugin, it's proven useful already in using ESLint with conjunction with our test suite.

I'm not really sure how to best say this because I'm new to unit testing altogether but normally we are writing our tests with a suite like this:

```
describe('DashboardView', function() {
    ...

    describe('on render', function() {
        ...
    });
});
```

We then have another set of tests in a different file for a different view:

```
describe('FiltersView', function() {
    ...

    describe('on render', function() {
        ...
    });
});
```

These are backbone views and we are testing some of their functionality.  If I turn on `no-suite-dupes` I get a warning for the FiltersView (which comes second).  It would be nice to have an option to have the dupes check on the same levels (so main level can't dupe, nested levels can't dupe, nested levels of nested levels can't dupe, etc).

Thoughts?

**Comment by ArnaudRinquin** [comment]
I definitely agree with this. In a large application is common to have the same `describe` names.

I'll see what I can do.

**Comment by tlvince** [comment]
@reintroducing are you finding that `no-suite-dupes` is triggering when the same suite name (`describe`) is used in different *files*?

I can reproduce this by writing duplicate suites in the same file, but not spanned over multiple files.

**Comment by reintroducing** [comment]
This was happening over different files. The two examples I gave above are from two separate files.
